tt_metal/common/constants.hpp

```diff
@@ -31,5 +31,11 @@ const double DEFAULT_PCC_THRESH = 0.999;
 constexpr uint32_t LF_MAN_PREC = 4;
 constexpr uint32_t FP16_MAN_PREC = 10;
 constexpr uint32_t FP16B_MAN_PREC = 7;
+
+// Empirically deduced di/dt problems appear for matmuls using more than 48 cores;
+// when there is 48 cores or less, we don't enable stagger used to mitigate the problems,
+// since the delay impacts op performance.
+// See issue #9857 for more info.
+constexpr uint32_t WH_B0_MM_MAX_CORES_NO_STAGGER = 48;
```

Comment:
This type of variable shouldn't be in shared tt_metal code, is it possible to push this definition to just llks?

Reply:
I don't think it is, not sure if in the llks you can get the info about how many other cores are involved, unless of course we pass it in? I can double check. This would also mean we would need to split the logic about whether we apply stagger or not in general between host and device.

Reply:
Is this used anywhere else? If not, you can just directly add it to `add_stagger_defines_if_needed`.